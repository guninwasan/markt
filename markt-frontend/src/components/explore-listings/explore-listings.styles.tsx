import styled from "styled-components";

const Container = styled.div`
  padding: 2rem;
`;

const Header = styled.h1`
  text-align: left;
  margin-bottom: 2rem;
`;

const SubheaderContainer = styled.div`
  display: flex;
  margin-bottom: 1rem;
  gap: 5rem;
  width: 70%;
`;

const Subheader = styled.h2<{ isSelected: boolean }>`
  cursor: pointer;
  font-size: 1rem;
  font-weight: normal;
  color: ${({ isSelected }) => (isSelected ? "black" : "#8F8F8F")};
  border-bottom: ${({ isSelected }) =>
    isSelected ? "2px solid black" : "none"};
  padding-bottom: 0.5rem;
`;

const ListingsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 1rem;
`;

const Listing = styled.div`
  border: 1px solid #ccc;
  padding: 1rem;
  width: calc(33.333% - 1rem);
  box-sizing: border-box;
`;

export {
  Container,
  Header,
  SubheaderContainer,
  Subheader,
  ListingsContainer,
  Listing,
};
