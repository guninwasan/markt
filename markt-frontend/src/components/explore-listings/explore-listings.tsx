import React, { useState } from "react";
import {
  Container,
  Header,
  SubheaderContainer,
  Subheader,
  ListingsContainer,
  Listing,
} from "./explore-listings.styles";

type SectionType = "featured" | "new" | "priceDrops";

type ListingsType = {
  [key: string]: string[];
};

const ExploreListings = () => {
  const [selectedSection, setSelectedSection] =
    useState<SectionType>("featured");

  const listings: ListingsType = {
    featured: [
      "Featured Listing 1",
      "Featured Listing 2",
      "Featured Listing 3",
      "Featured Listing 4",
    ],
    new: ["New Listing 1", "New Listing 2", "New Listing 3"],
    priceDrops: ["Price Drop 1", "Price Drop 2", "Price Drop 3"],
  };

  return (
    <Container>
      <Header>Explore Listings</Header>

      <SubheaderContainer>
        <Subheader
          isSelected={selectedSection === "featured"}
          onClick={() => setSelectedSection("featured")}
        >
          Featured Listings
        </Subheader>
        <Subheader
          isSelected={selectedSection === "new"}
          onClick={() => setSelectedSection("new")}
        >
          New to the Market
        </Subheader>
        <Subheader
          isSelected={selectedSection === "priceDrops"}
          onClick={() => setSelectedSection("priceDrops")}
        >
          Price Drops
        </Subheader>
      </SubheaderContainer>

      <ListingsContainer>
        {listings[selectedSection].map((listing, index) => (
          <Listing key={index}>{listing}</Listing>
        ))}
      </ListingsContainer>
    </Container>
  );
};

export { ExploreListings };
