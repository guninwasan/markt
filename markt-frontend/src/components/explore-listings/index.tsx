export { ExploreListings } from "./explore-listings";
