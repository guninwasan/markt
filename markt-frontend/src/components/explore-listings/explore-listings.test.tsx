import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ExploreListings } from "./explore-listings";

jest.mock("../listing-container", () => ({
  ListingContainer: () => <div>New Listing 1</div>,
}));

describe("ExploreListings", () => {
  it("should render the header", () => {
    render(<ExploreListings />);
    const headerElement = screen.getByText(/Explore Listings/i);
    expect(headerElement).toBeInTheDocument();
  });

  it("should render the subheaders", () => {
    render(<ExploreListings />);
    const featuredSubheader = screen.getByText(/Featured Listings/i);
    const newSubheader = screen.getByText(/Hot/i);

    expect(featuredSubheader).toBeInTheDocument();
    expect(newSubheader).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<ExploreListings />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
