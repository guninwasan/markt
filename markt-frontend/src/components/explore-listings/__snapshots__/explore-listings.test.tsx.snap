// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreListings should match the snapshot 1`] = `
<div
  class="sc-blHHSb cDukss"
>
  <h1
    class="sc-gtLWhw dmgHgk"
  >
    Explore Listings
  </h1>
  <div
    class="sc-egkSDF kdzCJS"
  >
    <h2
      class="sc-fAUdSK jcSlXs"
    >
      Featured Listings
    </h2>
    <h2
      class="sc-fAUdSK fLAMcP"
    >
      Hot
    </h2>
  </div>
  <div
    class="sc-dntaoT dQtacQ"
  />
</div>
`;
